src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -134,6 +134,13 @@ public void teleportHome(@NotNull Player player, @NotNull Island island, int del
         boolean trusted = getIslandTrusted(island, user).isPresent();
         boolean inIsland = user.getIsland().map(Island::getId).orElse(0) == island.getId();
         if (!island.isVisitable() && !inIsland && !trusted && !user.isBypassing()) {
+            if (island.isInIsland(player.getLocation())) {
+                if (user.getIsland().isPresent()) {
+                    teleportHome(player, user.getIsland().get(), delay);
```

Comment:
Well if the island is private, we dont want to send them to the island home

Reply:
Also this will result in an infinite loop?

Reply:
No, this will send the player to his own island, not the island he is currently trying to visit